### ISSUE TITLE
class property initializers

### ISSUE THREAD

**Issue by darklight9811** [denoland/deno_lint#609]
I actually came here with a doubt. Because deno doesn't allow you to create properties in classes without their initiacialization. And this is really preventing me from making my model package. I tried disabling it with tsconfig, but got lectured hard by some github users. And making the user have to disable their lint or anything else seems kinda dumb.

### Base Model
``` typescript
abstract class Model {
	// -------------------------------------------------
	// Properties
	// -------------------------------------------------

	// static
	protected static $table: string;
	protected static $fields: FieldInfoInterface[] = [];

	// instance
	private $values: Record<string, unknown> = {};

	// -------------------------------------------------
	// Constructor
	// -------------------------------------------------

	public constructor (fields: Record<string, unknown> = {}) {
		const $allFields = (this.constructor as unknown as {$fields: FieldInfoInterface[]}).$fields;

		// set fields
		for (let i = 0; i < $allFields.length; i++) {
			const field = $allFields[i];

			this[field.name as keyof this] = Object.defineProperty(this, field.name, {
				configurable: false,
				enumerable: false,
				set: (value) => {
					const onSet = dynamicTypes.get(field.type).onUpdate;
					this.$values[field.name] = onSet ? onSet(value, this.$values, field.args) : value;
				},
				get: () => {
					return this.$values[field.name];
				}
			});
		}

		// save values
		Object.keys(fields).forEach(key => {
			// deno-lint-ignore no-explicit-any
			this[key as keyof this] = fields[key] as any;
		});
	}

	// -------------------------------------------------
	// Main Methods
	// -------------------------------------------------

	public toObject () {
		return this.$values;
	}

	public toJson () {
		return JSON.stringify(this.$values);
	}

	// -------------------------------------------------
	// Query methods
	// -------------------------------------------------

	public static query () {
		return query().table(this.$table);
	}
}
```

### Model
``` typescript
class User extends Model {
	public id: string;
}
```

## Error
```
Property 'id' has no initializer and is not definitely assigned in the constructor.ts(2564)
```

## Nasty workarounds
- disable tslint on the file
- add a tsconfig (and risk breaking other packages)
- make the user add `!` after the field name (this is almost acceptable, but I'm really against forcing the user do this.

## Bonus
Adding experimental decorators to tsconfig also seems to break some packages, anything I could do about that?

**Comment by kitsonk** [comment]
> make the user add ! after the field name (this is almost acceptable, but I'm really against forcing the user do this.

This is the official way (for TypeScript) and has nothing to do with Deno lint. It is working as designed.
